### DISCUSSION TITLE
How does tailwind knows to include the correct classes form a popup menu?

### DISCUSSION THREAD

**Discussion by KarelJanVanHaute** [primefaces/community#1677] [PrimeReact] [answered]
When I test it in a contained environment, it works. The popup menu is correctly styled. But when I try to integrate it in my nextjs project, it doesn't seem to work. The components that are in my template get the correct styling. But the menu in popup mode gets injected into the DOM. And then the classes applied by default does not seem to work. 

Example of contained code: https://stackblitz.com/edit/48zhtr
What I get in my project:
![image](https://github.com/primefaces/community/assets/88485726/ec2c323b-01d1-431c-94dd-9a2433bec17c)

My Tailwind config:
```js
module.exports = {
  important: "#__next",
  content: [
    "./src/**/*.{js,ts,jsx,tsx}",
    "./node_modules/primereact/**/*.{js,ts,jsx,tsx}",
  ],
  darkMode: "selector"
};
```

My app:
```jsx
return (
    <PrimeReactProvider value={{ unstyled: true, pt: Tailwind }}>
      <Component {...pageProps} />
    </PrimeReactProvider>
  );
```

**Comment by melloware** [comment]
Please read the NextJs issue and its resolution: https://github.com/primefaces/primereact/issues/5187

**Comment by KarelJanVanHaute** [comment]
This doesn't really fix my problem. I get the correct styling on objects that are already in the DOM. But panels that get added to the DOM on an action don't get styled.

**Reply by melloware** [reply]
@KarelJanVanHaute can you create a StackBlitz reproducer showing the issue?

**Reply by KarelJanVanHaute** [reply]
@melloware I've created a minimal example here https://stackblitz.com/edit/stackblitz-starters-xbks7h?file=tailwind.config.ts

As you can see, the button gets the tailwind-classes applied, but the popup is totally unstyled.

**Reply by melloware** [reply, marked as answer]
@KarelJanVanHaute here is a working version:
[nextjs-unstyled-tailwind.zip](https://github.com/primefaces/community/files/15177159/nextjs-unstyled-tailwind.zip)

Unzip and run `npm install` then `npm run dev` to see it in action.

This is what your Menu looks in in Tailwind.

![image](https://github.com/primefaces/community/assets/4399574/d694c0f5-785d-4d43-aec8-947a02f2d794)

**Reply by KarelJanVanHaute** [reply]
@melloware What was I doing wrong? Also to help people out with the same problem.

**Reply by melloware** [reply]
i am not sure i just grabbed my starter kit from PrimeReact Examples.   Also i don't full trust StackBlitz when it comes to NextJS.
